specification/configuration.md

```diff
@@ -89,6 +90,50 @@ environment variables:
   If `SPLUNK_CONFIG` is defined then these options are not required but MAY be
   defined.
 
+### Kubernetes
+
+While Kubernetes supports container technology that can be configured using
+environment variables, package management solutions such as Helm charts and
+Operators require YAML-based configuration. As a result, Kubernetes package
+management solutions MUST support the following YAML configuration options:
+
+- `splunkCommon`
+  - `clusterName` ()          : [REQUIRED] Name of the cluster.
+  - `distro` ()               : Which distribution of Kubernetes is deployed.
+  - `environment` ()          : Name of the environment; if not defined then skipped.
+  - `logsEngine` (`fluentd`)  : How to collect logs.
+  - `modes`
```

Comment:
I don't agree that this is a `mode`. Helm can install the connector in many modes, so it's rather a component that can be enabled or disabled than a mode as in other single-component packaging solutions.

Reply:
In our docs today, we call it a mode: https://github.com/signalfx/splunk-otel-collector/blob/main/docs/agent-vs-gateway.md. I think we should be consistent. How about `deployment`?

Reply:
Can we keep "mode" term for single-component deployments and use something else like "component" multi-component deployments?

"deployment" doesn't seem to be a good choice to me because it can be confused with k8s deployment resource, and "agent" is a daemonset not deployment.

Looking at the customer experience so far the most common usage is adjusting recourses and adding extra components to the default agent's config. With the current helm chart, their custom helm values file would look like this:

```yaml
splunkObservability:
  accessToken: xxx
  realm: us0
otelAgent:
  resources:
    limits:
      cpu: 500m
      memory: 1Gi
  config:
    receivers:
      smartagent/docker-container-stats:
        type: docker-container-stats
        dockerURL: unix:///hostfs/var/run/docker.sock
    service:
      pipelines:
        metrics:
          receivers:
            - smartagent/docker-container-stats
            - hostmetrics
            - kubeletstats
            - otlp
            - receiver_creator
            - signalfx
```

It's already pretty deep nesting for agent configuration changes, and I heard about many cases when users confuse the layers. If we add 2 more levels on top of it (`splunkCommon` -> `modes`), I believe it'll make user experience worse.

And many of configurations options specific to helm chart release and kubernetes resources are on the top level because they already have established naming patterns in helm ecosystem, examples:

```
nameOverride: ""
fullnameOverride: ""
serviceAccount:
secret:
rbac:
```

Those options are pretty well known to those who used other helm chart before. So I think we should keep them at the top level. 

To summarize, I would really recommend to keep the components groups at the top level as well. May be we can prefix them with some common word, something like `componentAgent`, `componentGateway` and `componentK8sClusterReceiver`? I took `otel` prefix because we still have `fluent` as another component, so to keep the distinction now we have `otelAgent`, `otelCollector`, `otelK8sClusterReceiver` and `fluentd`. And for me it looks better than other alternatives, assuming `otelCollector` is renamed to `otelGateway`

Reply:
FWIW "mode" does sound a bit confusing to me as well. It sounds as if I can only pick one of the three modes especially if I'm using a single/common config file to deploy multiple agents.